### PR TITLE
Symfony 4 support

### PR DESCRIPTION
**Pull request by jorge07**

### CODE REVIEW

**Comment by cmodijk** [comment]
@jorge07 Thanks for taking the time to provide this pull request. What do you think should we wait until Symonfy 4.0 stable is releases to merge it?

**Comment by jorge07** [comment]
This PR it's a work in progress and will be until v4 mark as stable. But somehow I wanted to test the library with the new version and track the changes it need to make it work.

**Comment by cmodijk** [comment]
@jorge07 Yes, thanks !

**Comment by christophe-mailfert** [comment]
Need this as soon as you can :)

**Comment by jorge07** [comment]
Rebased

**Comment by jorge07** [comment]
New Symfony version require PHP >= 7. Should I keep 5.6 support?

**Comment by jorge07** [comment]
Squashed

**Comment by jorge07** [comment]
I thinks it's ready for review

File: composer.json [hunk 1]
```diff
@@ -19,10 +19,10 @@
     "require": {
         "php": "^5.6 || ^7.0",
         "simple-bus/message-bus": "~3.0",
-        "symfony/http-kernel": "~2.3 || ~3.0",
-        "symfony/dependency-injection": "~2.3 || ~3.0",
-        "symfony/yaml": "~2.3 || ~3.0",
-        "symfony/config": "~2.3 || ~3.0"
+        "symfony/http-kernel": "~2.7 || ~3.0 || ~4.0",
```
Comment (xabbuh) [inline composer.json]: If you take the chance to bump Symfony 2.x requirements to 2.7, I would also drop support for 3.0, 3.1, and 3.2.

File: composer.json [hunk 2]
```diff
@@ -31,12 +31,12 @@
...
     "require-dev": {
-        "phpunit/phpunit": "~4.0",
+        "phpunit/phpunit": "^5.7",
         "simple-bus/doctrine-orm-bridge": "~5.0",
         "doctrine/orm": "~2.3",
         "doctrine/doctrine-bundle": "~1.0",
-        "symfony/monolog-bridge": "~2.3 || ~3.0",
-        "symfony/monolog-bundle": "~2.3 || ~3.0"
+        "symfony/monolog-bridge": "~2.7 || ~3.0 || ~4.0",
+        "symfony/monolog-bundle": "~2.7 || ~3.0 || ~4.0"
```
Comment (xabbuh) [inline composer.json]: the release cycle of MonologBundle is independent from Symfony's release cycle (see https://github.com/symfony/monolog-bundle/releases)

**Comment by cmodijk** [comment]
I would say that we need support for;
* Al the LTS versions
    * 2.7 - 05/2018
    * 2.8 - 11/2018
    * 3.4 - 11/2020
    * 4.4 - 11/2022 (Not out yet)
* Currently supported by Symfony
    * ~~3.2 - 07/2017~~
    * 3.3 - 01/2018
* Development version(s)
    * 3.4 - 11/2020
    * 4.0 - 07/2018

Sources;
* https://symfony.com/roadmap
* https://symfony.com/doc/current/contributing/community/releases.html#version-history

**Comment by xabbuh** [comment]
3.2 does only receive security fixes until January 2018.

**Comment by cmodijk** [comment]
Can we also add support for tests with `composer update --prefer-lowest` as suggested in https://github.com/SimpleBus/SymfonyBridge/issues/51

File: composer.json (hunk 2 above)
Comment (jorge07) [reply composer.json]: Ops, true.

**Comment by cmodijk** [comment]
@xabbuh Thanks i updated the dates of my list and yes 3.2 can be dropped.

**Comment by cmodijk** [comment]
I'm closing this in favor of #58 thanks @jorge07 for the time

**Pull request closed without merging**